sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/SelectByteBuddyHelpers.java

```diff
@@ -174,10 +171,13 @@ static RowSelector createRowSelector(SchemaAndDescriptor schemaAndDescriptor) {
               .withParameters(Schema.class)
               .intercept(new SelectInstructionConstructor());
 
+      Optional<ClassLoader> rowClassLoader = Optional.ofNullable(Row.class.getClassLoader());
+      checkState(rowClassLoader.isPresent());
```

Comment:
You can get the same effect as your code without going through `Optional` by using `org.apache.beam.sdk.util.Preconditions.checkStateNotNull`. Here and in the other locations.

(I do prefer to use `Optional` in data structure design rather than null, because it is clearer and composes, but in this case it doesn't seem to add much)